feat(orders): add user orders list page with filters and delete funct…

dzx0902 opened the pull request:
## 描述

实现了用户订单列表页面，包含订单展示、筛选和删除功能。用户可以通过标签页切换查看不同状态的订单（全部订单、未出行、待支付），支持单程和往返订单的展示，并提供订单删除功能。

**主要功能：**
- 订单列表页面 (`/home/orders`)，支持三个筛选标签：全部订单、未出行、待支付
- 订单卡片展示，包含航班信息（航班号、出发/到达城市、时间）和乘客信息
- 软删除功能，仅允许删除已确认/已取消/已退款状态的订单
- 单程和往返订单的差异化显示
- 空状态提示和交互反馈
- 测试数据种子脚本，支持快速生成测试订单

## 相关问题

Resolved #165 

## 变更类型

- [ ] Bug 修复（不会破坏现有功能的非破坏性变更）
- [x] 新功能（添加功能的非破坏性变更）
- [ ] 破坏性变更（会导致现有功能无法正常工作的修复或功能）
- [ ] 文档更新
- [ ] 性能改进
- [ ] 代码重构
- [ ] CI/CD 改进

## 技术实现

### 新增文件：
- `src/app/(frontend)/(without-sidebar)/home/orders/page.tsx` - 服务端组件，负责认证和数据获取
- `src/app/(frontend)/(without-sidebar)/home/orders/page.client.tsx` - 客户端组件，负责 UI 交互
- `src/lib/queries/orders.ts` - 订单查询函数（getUserOrders, getUpcomingOrders, getPendingPaymentOrders）
- `src/lib/actions/orders.ts` - 新增 deleteOrderAction 服务端操作
- `src/types/actions/orders.ts` - 新增 OrderListItem 和 DeleteOrderResult 类型定义
- `scripts/seed-test-orders.ts` - 测试订单数据生成脚本
- `scripts/README.md` - 种子脚本使用文档

### 修改文件：
- `src/lib/schema/orders.ts` - 添加 Drizzle ORM 关系定义（ordersRelations, orderPassengersRelations, paymentsRelations）
- `src/lib/schema/flight-seat-classes.ts` - 添加 flightSeatClassesRelations
- `src/lib/schema/flights.ts` - 添加 flightsRelations（airline, departureAirport, arrivalAirport）
- `src/lib/schema/airports.ts` - 添加 airportsRelations（city）
- `package.json` - 新增 npm 脚本：`db:seed:orders`

### 核心功能：

1. **订单列表查询**
   - 使用 Drizzle ORM 关系查询，实现类型安全的嵌套数据获取
   - 过滤已删除订单（软删除）
   - 支持按状态和出发时间筛选

2. **软删除功能**
   - 验证订单所有权
   - 仅允许删除特定状态的订单（CONFIRMED/CANCELLED/REFUNDED）
   - 使用 deletedAt 字段实现软删除

3. **UI/UX 优化**
   - Tab 导航带计数显示
   - 订单状态徽章颜色区分
   - 删除确认对话框
   - Loading 状态和 Toast 提示
   - 响应式布局

4. **测试数据生成**
   - 支持为指定用户生成测试订单
   - 自动生成 6 种类型的订单，覆盖所有测试场景
   - 智能查询现有数据，避免硬编码

## 测试

- [x] 我已经在本地测试了这些更改
- [x] 种子脚本成功生成测试数据，覆盖以下场景：
  - 待支付订单（单程）
  - 已确认订单（单程，带增值服务）
  - 已确认订单（往返）
  - 已取消订单
  - 已退款订单
- [x] 订单列表正确显示，三个 Tab 筛选功能正常
- [x] 删除功能正常，状态验证生效
- [x] 我已经检查了代码能够无错误地构建（TypeScript 编译通过，ESLint 检查通过）

## 截图（如果适用）
1、tabs
<img width="1239" height="650" alt="{EAAB7310-48E9-4D55-ACDA-ECB6F60B9B72}" src="https://github.com/user-attachments/assets/24d1fcf3-618a-4da9-8b08-65fee90eb824" />
2、orders of different states
<img width="842" height="706" alt="{B57772FA-F88B-4112-952A-3A2C4EFC76BF}" src="https://github.com/user-attachments/assets/4ef61094-ef7e-4982-aed5-0c5b34ba6b01" />
<img width="852" height="620" alt="{C8BAACF7-C2E8-4635-90ED-BD6995A920BA}" src="https://github.com/user-attachments/assets/a042b1d6-e6c5-429c-807e-793aa9ee2028" />


## 检查清单

- [x] 我的代码遵循此项目的代码风格
- [x] 我已经对自己的代码进行了自我审查
- [x] 我已经对代码进行了注释，特别是在难以理解的地方
- [x] 我已经对文档进行了相应的更改（添加了种子脚本使用文档）
- [x] 我的更改没有产生新的警告
- [x] 已通过种子脚本验证功能的有效性
- [x] 新的和现有的单元测试在我的更改下都能通过

## 代码审查检查清单（供审查者使用）

- [ ] 代码可读且有良好的文档
- [ ] 更改经过了适当的测试
- [ ] 没有明显的性能问题
- [ ] 已解决安全考虑（订单所有权验证、软删除、输入验证）
- [ ] 破坏性变更已得到适当记录

## 附加说明

### 种子脚本使用方法：

**为第一个用户生成订单**：
```bash
pnpm run db:seed:orders

dzx0902 committed:
feat(orders): add user orders list page with filters and delete functionality

- Implement orders page with tab navigation (All/Upcoming/Pending Payment)
- Add order card display with flight info and passenger details
- Implement soft delete for orders with status check
- Add seed script to generate test order data
- Support both one-way and round-trip order display
- Add proper Drizzle ORM relations for type-safe queries

ukeSJTU commented on .claude/settings.local.json:
不要提交这些AI配置文件，我建议在 .gitignore 文件里面忽略这个文件夹

ukeSJTU commented on package.json:
请不要用dotenv这个依赖，Nextjs有自己的注入环境变量的方式，我记得dotenv有的时候会有兼容性问题。请用 next/env 这个第三方库，应该已经是项目依赖

ukeSJTU commented on src/lib/schema/airports.ts:
请解释一下为什么需要添加这个关系？

ukeSJTU commented on src/lib/schema/flight-seat-classes.ts:
请解释一下为什么要添加这个关系

ukeSJTU commented on src/lib/schema/flights.ts:
请解释一下为什么要添加这个关系

ukeSJTU commented on src/lib/schema/orders.ts:
请解释一下为什么要添加这个关系

dzx0902 committed:
chore: remove .claude from git tracking

dzx0902 commented on src/lib/schema/orders.ts:
1、启用关系查询：定义了 airportsRelations 后，你可以在查询时使用 with 来自动加载关联数据，例如：
```
// 查询机场时自动加载关联的城市信息
const airportsWithCities = await db.query.airports.findMany({
  with: {
    city: true,  // 这个 'city' 就是从 airportsRelations 定义来的
  },
});
```

2、在 src/lib/queries/orders.ts 中，我们就用到了这些关系定义：
```
const result = await db.query.orders.findMany({
  with: {
    outboundFlightSeatClass: {
      with: {
        flight: {
          with: {
            airline: true,
            departureAirport: {
              with: {
                city: true,  // ← 这里就用到了 airportsRelations 中定义的 city 关系
              },
            },
            arrivalAirport: {
              with: {
                city: true,  // ← 这里也用到了
              },
            },
          },
        },
      },
    },
  },
});
```
所以 relations 是被使用了的，它是 Drizzle ORM 关系查询功能的基础。没有这个定义，你就无法使用 with 进行嵌套的关系查询。

dzx0902 commented on package.json:
但是我这边不使用这个库的话，会如下的错误：
(base) PS F:\2025Autumn\CS3604\nomad> pnpm db:seed

> nomad@0.1.0 db:seed F:\2025Autumn\CS3604\nomad
> cross-env NODE_ENV=development tsx src/lib/db/seed.ts

F:\2025Autumn\CS3604\nomad\src\lib\db\index.ts:7
  throw new Error("DATABASE_URL environment variable is not set.");
        ^

Error: DATABASE_URL environment variable is not set.
    at <anonymous> (F:\2025Autumn\CS3604\nomad\src\lib\db\index.ts:7:9)
    at Object.<anonymous> (F:\2025Autumn\CS3604\nomad\src\lib\db\index.ts:54:2)
    at Module._compile (node:internal/modules/cjs/loader:1706:14)
    at Object.transformer (F:\2025Autumn\CS3604\nomad\node_modules\.pnpm\tsx@4.20.6\node_modules\tsx\dist\register-D46fvsV_.cjs:3:1104)
    at Module.load (node:internal/modules/cjs/loader:1441:32)
    at Function._load (node:internal/modules/cjs/loader:1263:12)
    at TracingChannel.traceSync (node:diagnostics_channel:328:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:237:24)
    at Module.require (node:internal/modules/cjs/loader:1463:12)
    at require (node:internal/modules/helpers:147:16)

Node.js v22.21.0
 ELIFECYCLE  Command failed with exit code 1.

ukeSJTU commented on package.json:
如果是 pnpm db:seed 那么应该看 .env.local 文件，如果是 pnpm db:seed:test 那么应该看 .env.test 文件里面有没有配置 DATABASE_URL 选项

ukeSJTU commented on src/lib/schema/orders.ts:
好的，明白了

dzx0902 committed:
fix(orders): remove the setting json of Claude and the dotenv

ukeSJTU committed:
feat(deps): remove dotenv dependency

We should use @next/env instead of dotenv for compatibility with Next.js framework.
